sway-core/src/error.rs

```diff
@@ -978,42 +970,7 @@ impl TypeError {
 
 impl CompileError {
     pub fn to_friendly_error_string(&self) -> String {
-        match self {
-            CompileError::ParseFailure { err, .. } => format!(
-                "Error parsing input: {}",
-                match &err.variant {
-                    pest::error::ErrorVariant::ParsingError {
-                        positives,
-                        negatives,
-                    } => {
-                        let mut buf = String::new();
-                        if !positives.is_empty() {
-                            buf.push_str(&format!(
-                                "expected one of [{}]",
-                                positives
-                                    .iter()
-                                    .map(|x| format!("{:?}", x))
-                                    .collect::<Vec<_>>()
-                                    .join(", ")
-                            ));
-                        }
-                        if !negatives.is_empty() {
-                            buf.push_str(&format!(
-                                "did not expect any of [{}]",
-                                negatives
-                                    .iter()
-                                    .map(|x| format!("{:?}", x))
-                                    .collect::<Vec<_>>()
-                                    .join(", ")
-                            ));
-                        }
-                        buf
-                    }
-                    pest::error::ErrorVariant::CustomError { message } => message.to_string(),
-                }
-            ),
-            a => format!("{}", a),
-        }
+        format!("{}", self)
     }
```

Comment:
Should we use [#[deprecated]](https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute) on this method recommending that we use `format!("{}")` like its implementation? Or maybe just remove it?

Reply:
Fair call.  I say remove it.
```
 : grep -Ir to_friendly_error_string .
./forc-pkg/src/pkg.rs:        .map(|e| e.to_friendly_error_string())
./sway-core/src/error.rs:    pub fn to_friendly_error_string(&self) -> String {
./sway-fmt/src/fmt.rs:                    e.to_friendly_error_string(),
./forc-util/src/lib.rs:    let friendly_str = maybe_uwuify(&err.to_friendly_error_string());
./sway-lsp/src/capabilities/diagnostic.rs:                message: error.to_friendly_error_string(),
```